rewrite-gradle/src/main/java/org/openrewrite/gradle/UpdateGradleWrapper.java

```diff
@@ -127,9 +129,15 @@ private GradleWrapper getGradleWrapper(ExecutionContext ctx) {
                 // services.gradle.org is unreachable, possibly because of a firewall
                 // But if the user specified a wrapperUri to an internal repository things might still be workable
                 if (wrapperUri == null) {
-                    throw new IllegalArgumentException(
-                            "Could not reach services.gradle.org and no alternative wrapper URI is provided. " +
-                            "To use this recipe in environments where services.gradle.org is unavailable specify a wrapperUri.", e);
+                    // If the user didn't specify a wrapperUri, but they did provide a specific version we assume they know this version
+                    // is available from whichever distribution url they were previously using and update the version
+                    if (Semver.validate(version, null).getValue() instanceof ExactVersion) {
+                        return gradleWrapper = new GradleWrapper(version, new DistributionInfos(null, null, null));
+                    } else {
+                        throw new IllegalArgumentException(
+                                "Could not reach services.gradle.org and no alternative wrapper URI is provided. " +
```

Comment:
Won't it be a good idea to maintain a constant file to store all these static messages ?